generated/src/aws-cpp-sdk-s3/source/S3ErrorMarshaller.cpp

```diff
@@ -96,4 +102,39 @@ Aws::String S3ErrorMarshaller::ExtractEndpoint(const AWSError<CoreErrors>& error
   }
 
   return {};
+}
+
+AWSError<Aws::Client::CoreErrors> S3ErrorMarshaller::Marshall(const Aws::Http::HttpResponse& httpResponse) const
```

Comment:
hm this isnt gunna work, by this logic the error would _always_ be in the body and we would never deserialize for a normal http error. i think you have to adopt your logic to look something like

```cpp
if (httpResponse.GetResponseCode() == HttpResponseCode::OK)
{
    // look in body
} else
{
    // run what is in XmlErrorMarshaller
}
```